src/Cli/dotnet/commands/CommonRunHelpers.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Build.Evaluation;
+using Microsoft.Build.Execution;
+using Microsoft.Build.Framework;
+using Microsoft.DotNet.Cli.Utils;
+
+namespace Microsoft.DotNet.Cli;
+
+internal static class CommonRunHelpers
+{
+    /// <param name="globalProperties">
+    /// Should have <see cref="StringComparer.OrdinalIgnoreCase"/>.
+    /// </param>
+    public static void AddUserPassedProperties(Dictionary<string, string> globalProperties, string[]? args)
+    {
+        Debug.Assert(globalProperties.Comparer == StringComparer.OrdinalIgnoreCase);
+
+        var fakeCommand = new System.CommandLine.CliCommand("dotnet") { CommonOptions.PropertiesOption };
+        var propertyParsingConfiguration = new System.CommandLine.CliConfiguration(fakeCommand);
+        var propertyParseResult = propertyParsingConfiguration.Parse(args);
+        var propertyValues = propertyParseResult.GetValue(CommonOptions.PropertiesOption);
+
+        if (propertyValues != null)
```

Comment:
Nit: revert condition to reduce nesting

Reply:
@Evangelink fwiw, it was a copy from `RunCommand`. But I'll do.